sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -271,57 +320,111 @@ case class TransformWithStateExec(
       case _ =>
     }
 
-    if (isStreaming) {
-      child.execute().mapPartitionsWithStateStore[InternalRow](
+    if (hasInitialState) {
+      val storeConf = new StateStoreConf(session.sqlContext.sessionState.conf)
+      val hadoopConfBroadcast = sparkContext.broadcast(
```

Comment:
Why do we need to do this ?

Reply:
I am not 100% percent sure, but this will distribute the read-only variable `hadoopConf` to all executors - similar as here: https://github.com/apache/spark/blob/74a9c6cfc6ea937031fe1ca5db539139322339a5/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateScanBuilder.scala#L55-L57

Reply:
Yeah there is a code comment. The practice seems to be that it's better to use broadcast rather than task serialization as it could be huge.